src/components/DataProvider/index.js

```diff
@@ -51,9 +51,10 @@ const getSubDomain = (baseDomain, subDomain) => {
   if (!subDomain) {
     return baseDomain;
   }
-  const minDomain = Math.max(subDomain[0], baseDomain[0]);
-  const maxDomain = Math.min(subDomain[1], baseDomain[1]);
-  return [minDomain, maxDomain];
+  const subDomainLength = subDomain[1] - subDomain[0];
```

Comment:
This seems like it will have all kinds of unforeseen consequences. I need to dig into this more to understand what's going to be impacted.

Reply:
Sorry for taking so long to review this. I put together some test cases (below), but I can't figure out how to make them run properly in griff so I'm just gonna copy the whole thing in here. Anyway, what do you think about the proposed implementation of `getSubDomain`? The current implementation has some known problems, and this PR's version doesn't correctly handle underflowing `subDomain`s.

```
import expect from 'expect';

// make sure that passed subDomain is part of baseDomain
const getSubDomain = (baseDomain, subDomain) => {
  if (!subDomain) {
    return baseDomain;
  }
  const baseDomainLength = baseDomain[1] - baseDomain[0];
  const subDomainLength = subDomain[1] - subDomain[0];
  if (baseDomainLength < subDomainLength) {
    return baseDomain;
  }
  if (subDomain[0] < baseDomain[0]) {
    return [baseDomain[0], Math.min(baseDomain[0] + subDomainLength)];
  }
  if (subDomain[1] > baseDomain[1]) {
    return [Math.max(baseDomain[1] - subDomainLength), baseDomain[1]];
  }
  return subDomain;
};

describe('DataProvider', () => {
  describe('getSubDomain', () => {
    it('handles the base case', () => {
      const baseDomain = [0, 100];
      const subDomain = [25, 75];
      expect(getSubDomain(baseDomain, subDomain)).toMatch(subDomain);
    });

    it('handles when the subdomain goes longer', () => {
      const baseDomain = [50, 100];
      const subDomain = [95, 105];
      expect(getSubDomain(baseDomain, subDomain)).toMatch([90, 100]);
    });

    it('handles when the subdomain goes shorter', () => {
      const baseDomain = [50, 100];
      const subDomain = [45, 55];
      expect(getSubDomain(baseDomain, subDomain)).toMatch([50, 60]);
    });

    it('handles when the subdomain is outside', () => {
      const baseDomain = [50, 100];
      const subDomain = [150, 160];
      expect(getSubDomain(baseDomain, subDomain)).toMatch([90, 100]);
    });

    it('handles when the subdomain is longer than the domain', () => {
      const baseDomain = [50, 100];
      const subDomain = [25, 125];
      expect(getSubDomain(baseDomain, subDomain)).toMatch([50, 100]);
    });
  });
});
```

If you put that into `tests/index-test.js` and run `yarn test` then it should pass. If you want to go for extra credit, you can figure out how to import `getSubDomain` from `DataProvider` and directly test it, as it should be.

Reply:
yeah you're right -- those don't make any sense as-written. I think what I meant to do was eg: `return [baseDomain[0], Math.min(baseDomain[0] + subDomainLength, baseDomain[1])]`

Basically, the math should never push the bounds of `baseDomain`

Reply:
right, but since `subDomainLength` is guaranteed to be <= to `baseDomainLength` at this point, won't `baseDomain[0] + subDomainLength` always be <= `baseDomain[1]`?